### ISSUE TITLE
vint throws an error when using escaped characters

### ISSUE THREAD

**Issue by netei** [Vimjas/vint#146]
With the following minimal vimrc : 

`nnoremap a :normal 0f.dt(^[`

Where the `^[` character is the one that you get when pressing `<c-v><Esc>`
gives the following error.

`/home/user/.tvimrc:0:0: Cannot detect encoding (binary file?): /home/ldap/hipped/.tvimrc (see no reference)`

I know that they is another alternative to give `<Esc>` in viml, but it would be great if vint was able to read files with escaped sequences.

If you don't want to add handling of escaped sequences, maybe it would be great to show on which line the escaped characters are ? WIth the current implementation, the error always says line 0, column 0, even when the escape character is on another line.

**Comment by RianFuro** [comment]
Looks like an issue with chardet. I fed the file to chardet linewise and it gave me this:
```
/usr/lib/python2.7/site-packages/chardet/universaldetector.py:74: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  if aBuf[:3] == codecs.BOM_UTF8:
/usr/lib/python2.7/site-packages/chardet/universaldetector.py:77: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  elif aBuf[:4] == codecs.BOM_UTF32_LE:
/usr/lib/python2.7/site-packages/chardet/universaldetector.py:80: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  elif aBuf[:4] == codecs.BOM_UTF32_BE:
/usr/lib/python2.7/site-packages/chardet/universaldetector.py:83: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  elif aBuf[:4] == b'\xFE\xFF\x00\x00':
/usr/lib/python2.7/site-packages/chardet/universaldetector.py:89: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  elif aBuf[:4] == b'\x00\x00\xFF\xFE':
/usr/lib/python2.7/site-packages/chardet/universaldetector.py:95: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  elif aBuf[:2] == codecs.BOM_LE:
/usr/lib/python2.7/site-packages/chardet/universaldetector.py:98: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  elif aBuf[:2] == codecs.BOM_BE:
```

Looks like chardet doesn't know that specific byte pattern - and is also unable to convert that to unicode. So i went and dumped it - just the ^[ escape sequence gave me:
```
00000000  1b 0a                                             |..|
00000002
```

`1B0A` Sure is an odd byte sequence and from what I checked it's not caught by chardet. Will check if changing that fixes something.

**Comment by RianFuro** [comment]
Ok, actually their PyPi package is out of date. I removed the chardet package (that was installed as a dependency), cloned their [repository](https://github.com/chardet/chardet) and installed that version. With the repos version, the filetype gets detected properly.

**Comment by RianFuro** [comment]
Poked them at chardet/chardet#83

**Comment by netei** [comment]
Ok, thanks :-)

**Comment by Kuniwak** [comment]
Thanks!

**Comment by jayvdb** [comment]
So this is likely solved by upgrading the requirement to 3.0.0, which was released 12 April 2017.

**Comment by jayvdb** [comment]
And we have two PRs which do bump the requirement, https://github.com/Kuniwak/vint/pull/215 and https://github.com/Kuniwak/vint/pull/217 , but neither set the lower limit to 3.x

**Comment by polyzen** [comment]
There was [another issue](https://github.com/chardet/chardet/pull/107#issuecomment-301206527) since then, so 3.0.3 is the way to go.

**Comment by jayvdb** [comment]
I've used chardet ~= 3.0.2 as some distros will still have 3.0.2, but 3.0.3 will be used by default.
So any bugs in vint due to 3.0.2 will be because the distro hasnt upgraded, and pressure on the distro is the best solution.

**Comment by polyzen** [comment]
This still affects me (and presumably anyone using [true colors & tmux](https://github.com/vim/vim/blob/37c64c78fd87e086b5a945ad7032787c274e2dcb/runtime/doc/term.txt#L481-L495)) with [chardet >=3.0.3](https://github.com/chardet/chardet/pull/107#issuecomment-301206527). Could anyone confirm?

**Comment by polyzen** [comment]
Fixed by #248, but this may require manual intervention (use of `scriptencoding` in affected scripts).

**Comment by blueyed** [comment]
Closing as per previous comment.
FYI: https://pypi.org/project/vim-vint/0.4a3/ was released just now - hopefully we'll have 0.4 soon.
